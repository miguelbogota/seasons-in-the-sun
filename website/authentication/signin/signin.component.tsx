import { TextField } from '@app/ui/text-field';
import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';

import { type AppSigninCredentials, appSigninCredentialsSchema, useAuthentication } from '../state';

export function Signin() {
  const { signin } = useAuthentication();
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<AppSigninCredentials>({
    criteriaMode: 'all',
    resolver: zodResolver(appSigninCredentialsSchema),
  });

  const onSubmit = handleSubmit(async (data) => {
    await signin(data).catch(console.error);
  });

  return (
    <div className="signin">
      <h2>Welcome Back</h2>
      <form onSubmit={onSubmit}>
        <TextField
          {...register('username')}
          label="Username"
          placeholder="Ex. johndoe"
          hint={errors.username?.message}
        />

        <TextField
          {...register('password')}
          label="Password"
          type="password"
          hint={errors.password?.message}
        />

        <button className="button" type="submit">
          Sign in
        </button>
      </form>
    </div>
  );
}
