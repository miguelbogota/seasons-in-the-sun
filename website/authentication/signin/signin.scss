@mixin apply {
  & .signin {
    display: block;
  }
}
