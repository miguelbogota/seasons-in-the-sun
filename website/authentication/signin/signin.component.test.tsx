import { render } from '@testing-library/react';

import { Signin } from './signin.component';

describe('Website / Authentication / Signin / Component', () => {
  beforeAll(() => {
    global.GUN = () => ({
      // @ts-expect-error Mocking get method.
      get: () => ({}),
      user: () => ({
        // @ts-expect-error Mocking get method.
        recall: () => ({}),
      }),
      on: () => ({}),
    });
  });

  it('should match snapshot', () => {
    const { container } = render(<Signin />);
    expect(container).toMatchSnapshot();
  });
});
