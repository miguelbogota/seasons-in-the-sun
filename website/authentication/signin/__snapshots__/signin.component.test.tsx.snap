// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Website / Authentication / Signin / Component > should match snapshot 1`] = `
<div>
  <div
    class="signin"
  >
    <h2>
      Welcome Back
    </h2>
    <form>
      <div
        class="text-field"
      >
        <input
          id=":r0:"
          name="username"
          placeholder="Ex. johndoe"
        />
        <label
          for=":r0:"
        >
          Username
        </label>
      </div>
      <div
        class="text-field"
      >
        <input
          id=":r1:"
          name="password"
          placeholder=""
          type="password"
        />
        <label
          for=":r1:"
        >
          Password
        </label>
      </div>
      <button
        class="button"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </div>
</div>
`;
